wallet-new/test/Integration/Expectations.hs

```diff
@@ -0,0 +1,29 @@
+module Integration.Expectations (MonadIntegrationExpectations(..)) where
+
+import           Universum
+
+import           Cardano.Wallet.Client.Http
+
+import           Integration.Util (Resp', asksM, fromResp)
+
+import qualified Test.Hspec as Hspec
+
+
+class MonadIntegrationExpectations m where
+    addressShouldExists :: Resp' WalletAddress -> m ()
+
+instance MonadIntegrationExpectations (ReaderT (WalletClient IO) IO) where
+    addressShouldExists resp = do
+        addr <- fromResp resp
+        addrs <- asksM getAddressIndex >>= fromResp
+        [ addrId | WalletAddress{..} <- addrs] `shouldContain` [addrId addr]
```

Comment:
The idea here, like in `Integration/Fixtures` is to define a clear DSL of actions we can use in the integration specs. This, we keep the integration specs quite simple and readable, and have the internals implemented here. 

I believe every assertion could be something in the form of `Resp' a -> m()` (or similar), making them convenient to use from the specs.

Reply:
I'm not really a fan of list comprehensions, to be honest -- ``map addrId addrs `shouldContain` [addrId addr]`` is more readable to me.